packages/block-editor/src/hooks/background.js

```diff
@@ -181,8 +177,44 @@ export function BackgroundImagePanel( {
 	);
 }
 
+function useBlockProps( { name, style } ) {
+	const { backgroundImageURL } = useSelect(
+		( select ) => {
+			const { getThemeFileURI } = unlock( select( blockEditorStore ) );
+			let file;
+			if (
+				!! style?.background?.backgroundImage?.url &&
+				style?.background?.backgroundImage?.source === 'theme'
+			) {
+				file = getThemeFileURI( style.background.backgroundImage.url );
+			}
+			return {
+				backgroundImageURL: file,
+			};
+		},
+		[ style?.background?.backgroundImage ]
+	);
+
+	if (
+		! hasBackgroundSupport( name, 'backgroundImage' ) ||
+		! backgroundImageURL
+	) {
+		return;
+	}
+
+	return {
+		style: {
+			// @TODO this should be backgroundImage. How to do that?
+			// Also, maybe consider reinstating https://github.com/WordPress/gutenberg/blob/fc98542a7dbba194bb4096d49cd0bd093b63f43e/packages/block-editor/src/hooks/background.js#L82
+			background: `url( '${ encodeURI(
```

Comment:
Quick question for @noisysocks or @ellatrix 

How can I update the value of a style property via `useBlockProps` or another hook? 

I want to overwrite `backgroundImage`, but I'm guessing the `useBlockProps` won't overwrite already-set props, only add new ones.... 🤔 

Thank you! 🙇🏻

Reply:
Oh never mind 😆 

I finally worked it out it's the filter order:

https://github.com/WordPress/gutenberg/blob/a8714f59987b1b252aa4fe5670bd164b7be7958d/packages/block-editor/src/hooks/index.js#L50